### ISSUE TITLE
ariaNotify API

### ISSUE THREAD

**Issue by keithamus** [mozilla/standards-positions#1049]
## Request for Mozilla Position on an Emerging Web Specification

* Specification title: ariaNotify API
* Specification or proposal URL (if available): https://github.com/w3c/aria/pull/2211
* Explainer URL (if available): https://github.com/WICG/accessible-notifications
* Proposal author(s) (`@`-mention GitHub accounts):  @douggeoffray, @keithamus
* Caniuse.com URL (optional): 
* Bugzilla URL (optional): 
* Mozillians who can provide input (optional): @jcsteh 
* WebKit standards-position: https://github.com/WebKit/standards-positions/issues/370

### Other information

This proposal is being actively prototyped in Chrome.

**Comment by jcsteh** [comment]
Overall, we believe that this will be a positive development for the web, especially given the challenges with the existing mechanism of ARIA live regions. However, there are [a number of substantive issues that need to be resolved in the specification](https://github.com/w3c/aria/issues?q=is%3Aissue+is%3Aopen+%5BAriaNotify%5D) before we can fully evaluate its impact. For now, we are deferring final assessment of our position until these issues have been resolved.

**Comment by janewman** [comment]
Given feedback from developer trials, we are looking toward shipping this feature in Chromium. @jcsteh, do you have any updated thoughts or positions on the ariaNotify API?

**Comment by jcsteh** [comment]
Great to hear that the origin trial went well. While I still think this is positive overall (as noted in https://github.com/mozilla/standards-positions/issues/1049#issuecomment-2359732118), I see that there are still [several open issues against ariaNotify](https://github.com/w3c/aria/issues?q=is%3Aissue+is%3Aopen+%5BAriaNotify%5D). Looking briefly, there have been some updates on these, but it's unclear what the findings were (whether from the original trial or discussions with stakeholders) and how these will be resolved (or whether impacted parts of the [proposed spec](https://github.com/w3c/aria/pull/2211) will be deferred to a future update). I think some clarity is needed on the direction here before I can reasonably make a formal position assessment. Thanks.

**Comment by jcsteh** [comment]
Thanks for these answers. This all seems reasonable to me.

> The following issues are related to features that will not be included in the first version of the API and are not included in the chromium implementation we are wanting to ship nor the spec draft.

As a more general question, is the Chromium implementation you are planning to ship completely consistent with [this spec pull request](https://github.com/w3c/aria/pull/2211); i.e. no more, no less, no differences? If so, I'll take another look at the pull request to ensure there are no further issues that haven't been filed. Thanks.

**Comment by janewman** [comment]
I have since created a new spec PR that builds off of Keith's work, https://github.com/w3c/aria/pull/2577
Outside of some clarifications and fixes to broken references, the only major addition in this updated PR is a new permission policy that allows top-level content to opt iframes out of aria notify. See [iframes and use in subresources section in the explainer for details on this behavior](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Accessibility/AriaNotify/explainer.md#iframes-and-use-in-subresources). This came out of the following conversation thread: https://github.com/whatwg/html/issues/11004

Regarding the chromium implementation, the API's web-facing functionality is entirely consistent with the spec PR, but there is a small divergence regarding the platform-specific implementation: 
There are a couple of circumstances where older platforms do not provide a reasonable notification API, in these circumstances chromium will generate a live region and fire live region events to provide the expected experience. These nodes are **only** in the tree we expose to screen readers and are not detectable in any way from web content. This behavior only occurs on older versions of Linux (atk < 2.50.0) and on windows chromium when the UIA implementation is not enabled.
I didn't feel that this fallback behavior belonged in the spec, but if you feel otherwise, please feel free to comment in the spec PR.

Here are the relevant changes in chromium in case you wanted to dig deeper:
* [6349538: [AriaNotify] Fire ARIA notifications for older versions of ATK/Linux](https://chromium-review.googlesource.com/c/chromium/src/+/6349538)
* [6663991: Reland "[AriaNotify] Add IA2 fallback for AriaNotify"](https://chromium-review.googlesource.com/c/chromium/src/+/6663991)
* [6711952: [AriaNotify] Add Policy to allow authors to block ariaNotify in iframes](https://chromium-review.googlesource.com/c/chromium/src/+/6711952)

Let me know if you have any further questions, happy to answer them!

**Comment by jcsteh** [comment]
@janewman, thanks for addressing my questions and for the new spec PR. All of my substantive concerns have been addressed.

We believe that this will be a positive development for the web. Compared to live regions, it significantly reduces developer friction and makes it much more intuitive for authors to implement accessible experiences which require text notifications, which should ultimately improve user experience. I am thus proposing a position of positive.

For absolute clarity, note that this position assessment relates purely to the v1 API proposed in w3c/aria#2577. Some of the proposed future additions to the API (out of scope for v1) still have unresolved questions and may substantively impact developer and user experience, so these will need to be assessed separately. That said, I appreciate the careful attention to concerns raised thus far and I'm sure we can work over time to resolve concerns for future additions.
